### ISSUE TITLE
Toggle threads / Disable threads

### ISSUE THREAD

**Issue by alexmingoia** [vector-im/element-meta#1090]
#### What would you like to do?

Disable threads, showing all messages in chronological order, as it was before the option to disable threads was removed in latest MacOS client.

#### Why would you like to do it?

Threads prevent me from being able to glance at the latest messages in a chat room. There is no longer a way to see all messages in a room in chronological order in a single view. This is a UX regression for me – not an improvement. The option to disable threads appears to have been removed.

#### How would you like to achieve it?

By restoring the option to disable threads.

**Comment by asteven** [comment]
Big honkin' +1 for a way to disable threads.
For me it's one of the worst UX features I've seen in a while.

**Comment by paulmelis** [comment]
Adding my own 2 eurocents. It's not that I want to completely disable threads, keeping the discussion structured is fine. But showing them in a separate pane when clicking on a collapsed thread in the main pane I find unusable, UX-wise. 

As echoed above (but somewhat different) I'd like a toggle to disable thread collapse and simply list all messages in chronological order. Messages within a thread might then be marker by thread title (or something like that), so you can still navigate to a thread-only view. I would even suggest you either show the main conversation (all messages), or only a single thread's messages, in the same pane. Using multiple panes for conversations is a pain (pun intended).

Btw, what is the "View in room" option on a thread supposed to do? I don't see any effect, while I would expect the thread to be shown inline in the main conversation view.

FWIW, I also hate the threading in discord and teams ("conversations"), it's just not an easy thing to support well in a UI.

**Comment by clokep** [comment]
> Btw, what is the "View in room" option on a thread supposed to do? I don't see any effect, while I would expect the thread to be shown inline in the main conversation view.

It brings you to the initial message of the thread in the main room timeline.

**Comment by paulmelis** [comment]
> > Btw, what is the "View in room" option on a thread supposed to do? I don't see any effect, while I would expect the thread to be shown inline in the main conversation view.
> 
> It brings you to the initial message of the thread in the main room timeline.

So that assumes that I'm reading a thread, but somehow have also scrolled the main view to a different point. I don't fully see that happening, as the reason to focus on a thread is to ignore everything else in the room, but okay.

**Comment by clokep** [comment]
> > > Btw, what is the "View in room" option on a thread supposed to do? I don't see any effect, while I would expect the thread to be shown inline in the main conversation view.
> > 
> > 
> > It brings you to the initial message of the thread in the main room timeline.
> 
> So that assumes that I'm reading a thread, but somehow have also scrolled the main view to a different point. I don't fully see that happening, as the reason to focus on a thread is to ignore everything else in the room, but okay.

If you have a long running thread it can be very useful to jump back to the context of when that thread started.

**Comment by asteven** [comment]
I liked the threading that Mattermost does.
It was intuitive to use and did not get in the way.

**Comment by hrenard** [comment]
Until this https://github.com/vector-im/element-web/issues/24392 is done, Threads are causing me way too many distractions. Disabling them was solving the issue, please reintroduce it.
For now, I switched clients (Cinny on desktop and Fluffy on mobile) to be able to focus and still be notified, that's a shame.

**Comment by axiopaladin** [comment]
+1 agreed. This should be a toggle in the `All Settings > Preferences` menu.

**Comment by setpill** [comment]
Making threads mandatory continues to be a horrible UX decision from the Element team. Please, for sake of accessibility and ease of use, make them optional again.

**Comment by joshuaboniface** [comment]
Yes please. Beyond being implemented in the worst conversation-breaking way possible, the fact they cannot be toggled off and, worse, even if they are toggled off, others can still use them (and thus show up as missing messages in rooms unlessI scroll all the way back to find the parent message), is beyond frustrating.

I'd go further: I want a way to turn them off entirely for rooms on my homeserver, or the way to choose how they're displayed so they are not intrusive bothers creating forked subrooms.

**Comment by ShinobiX9X** [comment]
I thought I'd like it, but in the group, there is 1 person that uses threads for every reply in the room. And now I just hate it and wish it never existed.

Please let me turn this very annoying feature off!!

**Comment by ErickRobertson** [comment]
How do we get this issue added to "Tasks Not Blocked by Spec Work" on this page?

https://github.com/vector-im/element-web/issues/24392

**Comment by t3chguy** [comment]
@ErickRobertson the Product team asking for it to be done. Right now they have not signed off on any way to disable threads.

**Comment by rghedin** [comment]
I just started using Matrix, it's awesome, but threads is… really bad. I'm surprised there's not an option to disable them.

**Comment by Tuerai** [comment]
I have been using element since it was riot, and this is the only change that has ever made it seem dramatically worse to me. I would very much like to disable the current view of threads and go back to how it was before.

**Comment by Nono-m0le** [comment]
count me in. I have huge accessibility issue with thread. PLEASE make it optionnal, it's a real disaster to follow communication and answers via threads ...

**Comment by markwylde** [comment]
I think while we're trying to solve the notification issues it would be good to allow disabling threads, or giving this issue some attention:
https://github.com/vector-im/element-web/issues/24392

With notifications jumping all over the place and possibly being hidden inside threads, it's just too easy to get lost in even small channels.

**Comment by jaakidup** [comment]
I agree with disabling threads, or at least give an option to disable it. 
Usability is a big decider for me, and I can't give my go ahead for clients to use element this way. 

It's nearly impossible to see which threads have replies, thus people miss conversations altogether. 
There is also an issue where the little dot notification that signals an unread thread or post in thread won't go away after reading it. 

I will also have to start using a different client.

**Comment by kilx** [comment]
Yes please, it's a terrible UX as it hides messages and I have to search for them. Usually I also discover that there is something in need of finding only days later.

On the Linux desktop client at least there's also a bug that makes all read threads be marked as unread at some point. Even ones from years ago. And the only way to mark them read is to go through them one by one only for them to be marked unread again after a restart.

I only have experience with threads in Element and Slack and both introduce more problems than issues they solve. The idea does have potential, but the UX I've seen so far are making things worse, not better.

If I have to do extra clicks and actively look for messages then it's a step back.

**Comment by sandstrom** [comment]
Somewhat related: my preference for threads would be to support what Twist does.

In Twist, a room has top-level threads, with 0-levels of additional nesting. So basically, there are no top level messages in a room, only top-level threads.

Quite useful for a room such as 'team A', where people start a new topic, discuss it, and then close the topic when it's all finalized.

I'm not saying every room should be like this. Should be configurable. The stream of consciousness style rooms, where everyone is just writing (interleaved) is also sometimes useful. Depends a lot on how the room is used.

**Comment by setpill** [comment]
Sounds like a fine idea to be discussed in its own issue rather than going off-topic on this one.

**Comment by Vir4030** [comment]
Please, Please, Please, just let us disable this feature on a channel-by-channel basis.  That seems like an easy toggle switch would let us disable this feature if we hate it.  Then we won't hate Element instead.

**Comment by wiz** [comment]
Threads are terrible, this ruins matrix so bad

**Comment by Vir4030** [comment]
How can we bring more attention to this, so that it can be done?

**Comment by SignFinder** [comment]
Hi team,
Let us know if you are able to pay attention to this request with more than 60 likes.

**Comment by donnyxray** [comment]
I was wondering why messages on my iPhone weren't showing up on my Desktop. Turns out I was in a thread. Noticing, finding and opening these threads is very unintuitive. Please let us disable this.

IMHO some forums solve this in a better way: they let select a few messages and split those off into a separate forum topic. Matrix could let us split a thread into its own room.

**Comment by murchandamus** [comment]
The deficiencies of the threads implementation is by far the most annoying aspect of Element to me. I dislike how threads pull apart the conversation in the first place, but I keep seeing rooms as having unread messages, and when I open the room there is nothing new, there is just some thread from months or years ago that is now marked as unread, but doesn’t even have a new message. IF there were at least a way to only show threads with unread messages or if marking the room as read also marked the corresponding threads as read, but the only way to get rid of the notification I have found is to scroll back until you find which thread has a dot. Obviously the scroll pane also resets every single time, so you have scroll up pages of threads multiple times, because usually it’s not just one thread.

The "reply" feature as it had been implemented before threads already and the possibility to quickly open a new room with a few people for specific topics that would just sink to the bottom of your "recent" list, but was easy to find again later did all that I needed much better. Threads was a huge regression in UX for me.

**Comment by djmaze** [comment]
>  but I keep seeing rooms as having unread messages, and when I open the room there is nothing new, there is just some thread from months or years ago that is now marked as unread, but doesn’t even have a new message.

As far as I am concerned, these problems have completely vanished since 1.11.58 of element-web / -desktop (i.e. the switch to rust-based crypto) and after enabling the _Threads Activity Centre_ in labs. Especially the latter one finally allows for a decent user experience with threads IMO. But I accept that many people still dislike them in general.

**Comment by FrostyX** [comment]
For our company communication, we use Slack, and we use threads heavily. They took some getting used to but I went from hate them to love them very quickly. Element threads have worse UX (I can never figure out who pings me and where) but that's not the problem.

The biggest problem is that only a few Matrix clients support threads. From the clients listed here https://matrix.org/ecosystem/clients only 3 clients support them, 10 don't, and 10 don't mention them, meaning they probably don't support them either. This leads to many people replying outside of the thread to threaded messages making the conversation a complete mess that is impossible to keep up with.

The threads feature should somehow be configurable on the room or space level. At this moment, it would be beneficial if communities with people using diverse Matrix clients could disable the threads feature.

**Comment by AndreKR** [comment]
For threads to be usable, in my opinion thread messages would have to (optionally) appear in the normal timeline (like Zulip's _topics_) and it must be possible to move messages to threads (otherwise the initial conversation is missing from the thread).

Unless something changes in this regard I would very much like to disable threads.

**Comment by lewri** [comment]
While not a fault of Element; use of threads in rooms bridged to Discord appears to result in some... weird behaviour.
While they do get delivered wrt the OP (and reference a message via mention); they are just tacked on the end of the text channel. Replying back over the bridge means it can lose context though; as generally you wouldn't always want to mention the originating post for your response.

It would generally be a benefit too to bridged spaces to have a toggle to disable threads in this kind of instance, though.

**Comment by heyakyra** [comment]
For clarification, this is for room members to have an option to collapse/flatten the _view_ of messages, not the ability to disable the _use_ of threading at all in a room https://github.com/element-hq/element-meta/issues/271 (complimentary to enforcing threading https://github.com/element-hq/element-meta/issues/334)

**Comment by beckerr-rzht** [comment]
In my opinion, the way Mattermost displays threads is ideal:

* The individual messages appear in the timeline.
* If you select the thread view for one of these entries, a panel opens on the right-hand side showing the entire thread.

This is a successful compromise between the timeline, where you can see messages as they arrived, and the thread view, which shows the thematic context.

**Comment by fallen** [comment]
> In my opinion, the way Mattermost displays threads is ideal:
> 
>     * The individual messages appear in the timeline.
> 
>     * If you select the thread view for one of these entries, a panel opens on the right-hand side showing the entire thread.
> 
> 
> This is a successful compromise between the timeline, where you can see messages as they arrived, and the thread view, which shows the thematic context.

And also, very importantly, Mattermost allows to follow or unfollow a thread.
So that you won't get 50 notifications if that threads gets 50 replies but you don't care about this particular topic.
In Element, you can't unfollow threads and you therefore get tons of notifications.
I'm coming from Mattermost and I see a BIG difference in the time I spend reading messages. Messages that I was not reading before.

**Comment by JBRDLR** [comment]
Please make it look like in MatterMost or provide the option to design your self a similar view :rocket:
